nixos/modules/system/boot/stage-2-init.sh

```diff
@@ -54,7 +54,7 @@ if [ ! -e /proc/1 ]; then
 fi
 
 
-if [ "${IN_NIXOS_SYSTEMD_STAGE1:-}" = true ]; then
+if [ "${IN_NIXOS_SYSTEMD_STAGE1:-}" = true ] || [ -n "$container" ] ; then
```

Comment:
I think it would make more sense to check for the non-existence of /dev/kmsg here.  A container is one case where there's no /dev/kmsg, but if there are others, we'd want to do the same thing here, right?

Reply:
I've initially started by checking for `/dev/kmsg`, a bit differently. I agree it will cover all cases where `/dev/kmsg` is missing.